lib/cn/cn_tree.c

```diff
@@ -1530,6 +1553,16 @@ cn_comp_update_split(
     rmlock_wlock(&tree->ct_lock);
 
     do {
+
+        /* If a root spill added a kvset to this node in the middle
+         * of the split operation then we must abort and retry.
+         */
+        if (w->cw_kvset_cnt != cn_ns_kvsets(&w->cw_node->tn_ns)) {
+            err = merr(EAGAIN);
+            ev_debug(1);
+            break;
+        }
```

Comment:
Can't throw errors in this function.

Reply:
Right, was following the lead of cn_comp_update_split() which recently passed muster...